openff/toolkit/topology/molecule.py

```diff
@@ -362,6 +363,35 @@ def partial_charge(self):
             index = self.molecule_atom_index
             return self._molecule._partial_charges[index]
 
+    @partial_charge.setter
+    def partial_charge(self, charge):
+        if self.molecule.partial_charges is None:
+            raise MissingPartialChargesError(
+                "Cannot set individual atom's partial charge if it is in a molecule with no partial charges. "
+                "Instead, use the `Molecule.partial_charges` setter. If this behavior is important to you, "
+                "please raise an issue describing your use case."
+            )
+
+        if not isinstance(charge, (unit.Quantity, int, float)):
```

Comment:
(blocking) We should disallow `int` here to avoid users confusing this for `Atom.formal_charge`